_posts/2022-10-27-Adding-New-Distributions-to-OpenSearch-Project.md

```diff
@@ -0,0 +1,411 @@
+├── run_perf_test.py
+├── run_sign.py
+├── sign_workflow
+├── system
+└── test_workflow
+```
+
+The corresponding directories: `build_workflow`, `assemble_workflow`, and `test_workflow` contain the code for the `build`, `assemble`, `test` stages, respectively. In this blog, we focus on the `build` and `assemble` stages, as they are crucial to implementing a new OpenSearch distribution.
+
+While we are using Jenkins to execute the opensearch-build repo code for automated distribution builds, this blog doesn’t go into depth on that topic; see [OpenSearch automated build system is now live](https://opensearch.org/blog/community/2022/10/public-jenkins/) for more information. In this blog, we focus on running the `build` and `assemble` code on a local environment with the new distribution implemented with necessary details. 
+
+**Example: Simplified flow to generate artifacts on Jenkins**
+<img src="/assets/media/blog-images/2022-10-27-adding-new-distributions-in-opensearch-project/jenkins_pipeline_flowchart.png" alt="Jenkins Pipeliens Flowchart"/>{: .img-fluid }
+As shown in the figure above, after contributors commit their code or merge pull requests (PRs) in [opensearch-project](https://github.com/opensearch-project) repo, the pipelines on our [Public Jenkins](https://build.ci.opensearch.org/) will be triggered. From there, `build_workflow`, `assemble_workflow`, and `test_workflow` will be run sequentially and then the final products will be promoted to the object storage production bucket and made available publicly at `artifacts.opensearch.org`.
+
+## How to run build_workflow
+
+A common question asked during the `build_workflow` is “How do we run it?”. To answer that question, it’s necessary to understand what available parameters we have access to control the output.
+
+`Older version of opensearch-build/src/build_workflow/README.md before we add RPM`
+
+```
+The following parameters are available in `build.sh`.
+
+| name                    | description                                     |
+|-------------------------|-------------------------------------------------|
+| -s, --snapshot          | Build a snapshot instead of a release artifact. |
+| -a, --architecture      | Specify architecture to build.                  |
+| -p, --platform          | Specify platform to build.                      |
+| --component [name ...]  | Rebuild a subset of components by name.         |
+| --keep                  | Do not delete the temporary working directory.  |
+| -l, --lock              | Generate a stable reference manifest.           |
+| -v, --verbose           | Show more verbose output.                       |
+
+```
+
+The figure above shows available parameters in `build_workflow`, before `RPM` is implemented. We do not have a `--distribution` parameter, as we were using `--platform` parameter to hardcode the distribution type. Initially, we assumed if `platform=linux, distribution=tar`. However, `RPM` also runs on LINUX hosts. 
```

Comment:
Not really sure how to make it better.
What I want to describe is we used to assume `distribution` type based on `platform` type.
But that is not true anymore since we introduce RPM, because RPM can also run on linux platform.

Therefore, we need a new `--distribution` parameter to explicitly define the distribution type.

Reply:
The figure above shows the parameters available in `build_workflow` before `RPM` is implemented. There is no `--distribution` parameter, as the `--platform` parameter is used to hardcode the distribution type. Initially, it was assumed that if `platform=linux`, then `distribution=tar`.

Reply:
a